core/modules/replica/SqlResultSet.cc

```diff
@@ -208,13 +206,96 @@ util::ColumnTablePrinter SqlResultSet::toColumnTable(string const& caption,
 }
 
 
+void SqlResultSet::set(ProtocolResponseSql const& message) {
+    queryResultSet.clear();
+    for (int i = 0; i < message.result_sets_size(); ++i) {
+        auto&& resultSetMessage = message.result_sets(i);
+        queryResultSet.emplace(resultSetMessage.scope(), resultSetMessage);
+    }
+}    
+
+
+json SqlResultSet::toJson() const {
+
+    json resultJson = json::array();
+
+    for (auto&& itr: queryResultSet) {
+        auto&& scope = itr.first;
+        auto&& resultSet = itr.second;
+        json resultSetJson = json::object();
+        resultSetJson["scope"] = scope;
+        resultSetJson["result_set"] = resultSet.toJson();
+        resultJson.push_back(resultSetJson);
+    }
+    return resultJson;
+}
+
+
+bool SqlResultSet::hasErrors() const {
+    for (auto&& itr: queryResultSet) {
+        auto&& result = itr.second;
+        if (result.extendedStatus != ExtendedCompletionStatus::EXT_STATUS_NONE) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
Not sure if the solution would fit into a single line (given LSST C++ Style Guide constraints):
```
    return not any_of(queryResultSet.cbegin(), queryResultSet.cend(),
            [desiredExtendedStatus](pair<std::string,ResultSet> const& elem) {
        return elem.second.extendedStatus != desiredExtendedStatus;
    });
```
Though, this form would be a bit harder to comprehend

Reply:
Since we are using C++14 you should be able to write a bit shorter version:

    return any_of(queryResultSet.cbegin(), queryResultSet.cend(),
            [](auto const& elem) {
                return elem.second.extendedStatus != ExtendedCompletionStatus::EXT_STATUS_NONE;
             });

I think this is more readable that original loop.

Reply:
FWIW, I find the algo+lambda to be significantly **easier** to comprehend than the explicit loop; as soon as I parse the `any_of` I know exactly what is intended, and the rest is idiomatic. You intention is declared, and I don't have to read through the entire loop before I can figure out what you are trying to do.

I format them like this, which I find quite compact and nice:

```
return any_of(queryResultSet.cbegin(), queryResultSet.cend(), [](auto const& elem) {
    return elem.second.extendedStatus != ExtendedCompletionStatus::EXT_STATUS_NONE;
});
```

Reply:
Thanks, folks! I wasn't just aware that C++14 allows generic lambdas. And this is really cool! That's quite handy. https://isocpp.org/wiki/faq/cpp14-language#generic-lambdas